File: backend/app/models/task.py
from datetime import date
from typing import Literal

from pydantic import BaseModel


TaskPriority = Literal[
    "high",
    "medium",
    "low",
]


TaskState = Literal[
    "▶️",
    "⏳",
    "🚧",
]


class Task(BaseModel):

    text: str
    path: str
    line: int

    completed: bool

    priority: TaskPriority | None = None
    deadline: date | None = None

    task_state: TaskState | None = None

    waiting_for: str | None = None
    waiting_since: date | None = None
    follow_up: date | None = None

    blocked_by: str | None = None

File: frontend/src/types/task.ts
export type TaskPriority =
  | "high"
  | "medium"
  | "low";


export type TaskState =
  | "▶️"
  | "⏳"
  | "🚧";


export type WorkloadLevel =
  | "Niedrig"
  | "Moderat"
  | "Hoch"
  | "Sehr hoch";


export interface Task {
  text: string;
  path: string;
  line: number;

  completed: boolean;

  priority: TaskPriority | null;
  deadline: string | null;

  task_state: TaskState | null;

  waiting_for: string | null;
  waiting_since: string | null;
  follow_up: string | null;

  blocked_by: string | null;
}


export interface TaskWorkload {
  days: number;

  deadline_count: number;
  high_priority_count: number;

  waiting_count: number;
  blocked_count: number;

  level: WorkloadLevel;
  score: number;
}